vars/spinnakerTriggerPipeline.groovy

```diff
@@ -0,0 +1,176 @@
+        keyFileCredentialsId: 'keyFileCredentialsId',
+        gateUrl: 'gateUrl',
+        pipelineParameters: 'pipelineParameters',
+        pipelineNameOrId: 'pipelineNameOrId'
+    ]
+]
+
+/**
+ * Triggers a [Spinnaker](https://spinnaker.io) pipeline from a Jenkins pipeline.
+ * Spinnaker is for example used for Continuos Deployment scenarios to various Clouds.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        final script = checkScript(this, parameters) ?: this
+
+        // load default & individual configuration
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults(CONFIG_KEY_COMPATIBILITY)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixin(parameters, PARAMETER_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .withMandatoryProperty('spinnaker/gateUrl')
+            .withMandatoryProperty('spinnaker/application')
+            .withMandatoryProperty('spinnaker/pipelineNameOrId')
+            .use()
+
+        // telemetry reporting
+        new Utils().pushToSWA([
+            step: STEP_NAME
+        ], config)
+
+        int timeoutMinutes = parameters.timeout ?: 60
+
+        String paramsString = ""
+        if (config.spinnaker.pipelineParameters) {
+            def pipelineParameters = [parameters: config.spinnaker.pipelineParameters]
+
+            paramsString = "-d '${new GStringTemplateEngine().createTemplate(JsonOutput.toJson(pipelineParameters)).make([config: config, env: env]).toString()}'"
+
+            if (config.verbose) {
+                echo "[${STEP_NAME}] Triggering Spinnaker pipeline with parameters: ${paramsString}"
+            }
+        }
+
+        def pipelineTriggerResponse
+
+        //ToDO: support userId/pwd authentication or token authentication!
+
+        def curlVerbosity =  (config.verbose==true) ? '--verbose ' : '--silent '
+
+        withCredentials([
+            file(credentialsId: config.spinnaker.keyFileCredentialsId, variable: 'clientKey'),
+            file(credentialsId: config.spinnaker.certFileCredentialsId, variable: 'clientCertificate')
+        ]) {
+            // Trigger a pipeline execution by calling invokePipelineConfigUsingPOST1 (see https://www.spinnaker.io/reference/api/docs.html)
+            pipelineTriggerResponse = sh(returnStdout: true, script: "curl -H 'Content-Type: application/json' -X POST ${paramsString} ${curlVerbosity}--insecure --cert \$clientCertificate --key \$clientKey ${config.spinnaker.gateUrl}/pipelines/${config.spinnaker.application}/${config.spinnaker.pipelineNameOrId}").trim()
```

Comment:
1. I guess it is okay to fail some lines later, especially since the error is basically the same: "cannot trigger pipeline"

2. we might be able to go without although custom certificates may create an issue ...

3. no, that is also my assumption ..

Reply:
Ad 1.) 

> I guess it is okay to fail some lines later

I disagree. Failing at the point were the issue is triggered or failing a few lines later might make a huge difference. When the stack trace starts at a line where xml, yml, json or &lt;whatever&gt; parsing is performed somebody might start with the assumtion that the format is invalid and investigate into that direction. But the problem is not an invalid format, but a connection issue that happend earlier. The content which is attempted to be parsed is in this case is e.g. the http body delivered alongside with e.g. an http-404, informing with some nice html that the resource cannot be found (e.g. a simple misconfiguration of an URL, or a proxy issue ...). It may take some time until the trouble-shooter finds the real issue, especially in case we have no direct and fast communication (you know, we have to ask back, the contact needs to ask a collegue first, we may have time-zone delays etc ...)

_By the way: most of our curl statement does not come with good proxy support, e.g. `--noproxy <url_goes_here>` but that is another story._